### PR TITLE
Fix Datepicker Validation to Allow Independent Date Updates

### PR DESCRIPTION
**Pull request by jayesh-unde**
Previously, users were unable to update only one date (either the start date or end date) ,if either date was already set to a value less than the current date. Then users were forced to update both dates simultaneously, even when only one needed to be changed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,5 +86,9 @@
     "vite": "^3.2.4",
     "vite-plugin-env-compatible": "^1.1.1",
     "vite-plugin-svgr": "^2.2.2"
-  }
+  },
+  "description": "[![Netlify Status](https://api.netlify.com/api/v1/badges/621ca722-aec5-4a86-958b-b6a24ef9c010/deploy-status)](https://app.netlify.com/sites/staging-iitp/deploys)",
```
Comment (Anuragtech02) [inline package.json:90]: Don't add this in the project

File: src/pages/Profile/index.tsx [hunk 2]
```diff
@@ -13,11 +13,13 @@ import { AuthContext } from "../../utils/auth/AuthContext";
...
 const { profilePlaceholder } = images;
 
+// line 225 return statement
+
 const Profile = () => {
   const { userId } = useParams();
 
   const [form] = Form.useForm();
-  const [user, setUser] = useState<UserType | null>(null);
+  const [user, setUser] = useState<UserType | null>(null);  // value is set in line 195 using useEffect
```
Comment (Anuragtech02) [inline src/pages/Profile/index.tsx:22]: What does this mean?

File: src/pages/Questions/Questions.tsx [hunk 3]
```diff
@@ -374,7 +374,7 @@ const Questions = () => {
         color={
           value === "Easy"
             ? "green"
-            : value.toLowerCase() === "medium"
+            : value?.toLowerCase() === "medium"
```
Comment (Anuragtech02) [inline src/pages/Questions/Questions.tsx:377]: Right a method for this instead or maybe create an object that returns this

File: src/pages/Test/CreateTest.tsx [hunk 4]
```diff
@@ -139,9 +139,12 @@ const CreateTest = () => {
           ...rest,
         }));
         console.log({ batches });
-        if (validity?.from && validity?.to) {
+        // validity?.from && 
+        if (validity?.from || validity?.to) {
+          console.log('Test Date Range:', [dayjs(validity.from), dayjs(validity.to)]);
```
Comment (Anuragtech02) [inline src/pages/Test/CreateTest.tsx:144]: Remove logs

File: src/pages/Test/CreateTest.tsx [hunk 5]
```diff
@@ -139,9 +139,12 @@ const CreateTest = () => {
...
           setTestDateRange([dayjs(validity?.from), dayjs(validity?.to)]);
         }
+        console.log('Test Date Range:', [dayjs(validity.from), dayjs(validity.to)]);
```
Comment (Anuragtech02) [inline src/pages/Test/CreateTest.tsx:147]: Same here

File: src/pages/Test/CreateTest.tsx [hunk 6]
```diff
@@ -599,9 +602,11 @@ const CreateTest = () => {
             >
               <CustomDateRangePicker
                 showTime={true}
-                onChange={(props: any) => setTestDateRange(props)}
+                onChange={(props: any) => {
+                  console.log("Answer is ",testDateRange);
+                  setTestDateRange(props)}}
                 value={testDateRange}
-                disablePrevDates={true}
+                disablePrevDates={true} // it should be true only
```
Comment (Anuragtech02) [inline src/pages/Test/CreateTest.tsx:609]: Booleans could be passed simply without having to explicitly write `true`
```suggestion
               disablePrevDates
```

**Review by Anuragtech02 — CHANGES_REQUESTED** [review]
Please look into the changes

File: src/pages/Profile/index.tsx (hunk 2 above)
Comment (jayesh-unde) [reply src/pages/Profile/index.tsx:22]: actually the comment was made for my understanding of the project

File: src/pages/Test/CreateTest.tsx (hunk 4 above)
Comment (jayesh-unde) [reply src/pages/Test/CreateTest.tsx:144]: Yes sir

File: src/pages/Test/CreateTest.tsx (hunk 6 above)
Comment (jayesh-unde) [reply src/pages/Test/CreateTest.tsx:609]: I have not made any changes here, the changes were made only in CustomDateRangePicker.tsx file

**Pull request merged**